From b8859311bceb0cd63a64c2fcc8967f0c6466d736 Mon Sep 17 00:00:00 2001
From: Franck Bui <fbui@suse.com>
Date: Wed, 19 Oct 2022 15:28:27 +0200
Subject: [PATCH] random-seed: make one more use of random_write_entropy()

No functional change.

(cherry picked from commit 141d1da021514be2cc7e7a903fa83b11f6054db6)

Related: RHEL-16952
---
 src/random-seed/random-seed.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/random-seed/random-seed.c b/src/random-seed/random-seed.c
index 569b916f4a..d94005bdde 100644
--- a/src/random-seed/random-seed.c
+++ b/src/random-seed/random-seed.c
@@ -197,7 +197,7 @@ static int run(int argc, char *argv[]) {
                 if (r < 0)
                         log_debug_errno(r, "Failed to get machine ID, ignoring: %m");
                 else {
-                        r = loop_write(random_fd, &mid, sizeof(mid), false);
+                        r = random_write_entropy(random_fd, &mid, sizeof(mid), /* credit= */ false);
                         if (r < 0)
                                 log_debug_errno(r, "Failed to write machine ID to /dev/urandom, ignoring: %m");
                 }
